src/pages/MyPatches.tsx

```diff
@@ -60,34 +69,55 @@ export const MyPatches = () => {
       } catch (e) {
         // empty block
       }
-    });
-  }, [networkStatus, error, fetchMore, listen]);
-
-  if (error) {
-    return <div>{error.message}</div>;
-  }
+    };
+    fetch();
+  }, [page, search, prevSearch, fetchMore]);
 
   const onCheckboxChange = () => {
     replace(
-      `${pathname}?${queryString.stringify({
-        ...queryString.parse(search),
-        [MyPatchesQueryParams.CommitQueue]: !getQueryVariables(search)
-          .$includeCommitQueue,
-      })}`
+      `${pathname}?${queryString.stringify(
+        {
+          ...queryString.parse(search, { arrayFormat }),
+          [MyPatchesQueryParams.CommitQueue]: !getQueryVariables(search)
+            .includeCommitQueue,
+        },
+        { arrayFormat }
+      )}`
     );
   };
 
+  const onChange = (pageNum: number) => {
+    setPage(pageNum);
+  };
+
+  const tableBody = (() => {
+    if (error) {
+      return <ErrorWrapper>{error}</ErrorWrapper>;
+    }
+    if (loading) {
+      return (
+        <StyledSkeleton active={true} title={false} paragraph={{ rows: 4 }} />
+      );
+    }
+    if (get(data, "userPatches.patches", []).length !== 0) {
+      return data.userPatches.patches.map((p) => (
+        <PatchCard key={p.id} {...p} />
+      ));
+    }
+    return <NoResults data-cy="no-patches-found">No patches found</NoResults>;
+  })();
```

Comment:
Instead of an IIFE you can just capitalize the first letter of the function and turn it into a jsx component. 
```
const TableBody = ({ error, loading, data }) => {
  if (error) {
    return <ErrorWrapper>{error}</ErrorWrapper>;
  }
  if (loading) {
    return (
      <StyledSkeleton active={true} title={false} paragraph={{ rows: 4 }} />
    );
  }
  if (get(data, "userPatches.patches", []).length !== 0) {
    return data.userPatches.patches.map((p) => <PatchCard key={p.id} {...p} />);
  }
  return <NoResults data-cy="no-patches-found">No patches found</NoResults>;
};
```

Reply:
I rather keep this as an IIFE because passing props to the component seems repetitive when the variables are available in the closure.

Reply:
OH now we using an IIFE??!!!?!??!?!???!?!?!?!?!?!